### PR TITLE
Upgrade branch 2.6.x using TemplateControl

### LINKED ISSUE #57: no jetty-alpn-agent-2.0.6.jar
when running ./play run, following error message shows:

> Error opening zip file or JAR manifest missing : /Users/twer/source/scala/play-scala-tls-example/target/universal/stage/jetty-alpn-agent/jetty-alpn-agent-2.0.6.jar

should I download the jar and put it to the specified place by myself? or do I missed some step?

### LINKED ISSUE #74: Add Java 11 to CI
One way of doing this is by adding the following to `.travis.yml`:

```yaml
matrix:
  include:
    - scala: 2.12.7
      jdk: openjdk11
  allow_failures:
    # Java 11 is still not fully supported. It is good that we are already
    # testing our sample applications to better discover possible problems
    # but we can allow failures here too.
    - jdk: openjdk11
```

Make sure that's the right Scala version.  Also we might want to use AdoptOpenJDK 11 instead of Oracle OpenJDK 11.

### PR DESCRIPTION
**Pull request by octonato**
This PR was originally triggered by template control but it evolves into a fix for many errors and improvements that were in place for some time:

* fixed cross compilations
* remove broken jdk9 build
* Resolves #74 - added openjdk11 build, but allowed to fail (see #74)
* Fixes #57

### CODE REVIEW

File: modules/one/project/build.properties [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #Activator-generated Properties
 #Fri Jul 08 22:56:54 PDT 2016
 template.uuid=f57599cf-a4f5-4333-9ff1-55d48fd49891
-sbt.version=1.1.0
+sbt.version=1.2.1
```
Comment (marcospereira) [inline modules/one/project/build.properties]: 1.2.6 is out.

File: scripts/script-helper [hunk 2]
```diff
@@ -5,6 +5,9 @@ set -o pipefail
 
 java_version=$(java -version 2>&1 | java -version 2>&1 | awk -F '"' '/version/ {print $2}')
 
+# Export the keystore password for use in ws.conf
+export KEY_PASSWORD=`cat scripts/password`
```
Comment (marcospereira) [inline scripts/script-helper]: 👍

**Comment by marcospereira** [comment]
@renatocaval, since you are removing oraclejdk9 you can remove all the `java_version` dance in `scripts/script-helper`.

**Comment by octonato** [comment]
@marcospereira, I added the dance for jdk11 instead. I also added a build for openjdk11 as suggested by @dwijnand [here](https://github.com/playframework/play-scala-tls-example/issues/74).

Can you give it a second review?

**Comment by octonato** [comment]
I blindly added `--add-modules=java.xml.bind` for Java 11 and the build is now failing with:

```
Picked up _JAVA_OPTIONS:  --add-modules=java.xml.bind
OpenJDK 64-Bit Server VM warning: Ignoring option MaxPermSize; support was removed in 8.0
Error occurred during initialization of boot layer
java.lang.module.FindException: Module java.xml.bind not found
```

Although we do expect Java11 build to fail, I was not expecting it to fail with this. 

I'm removing that "dance" for now.

File: play [hunk 3]
```diff
@@ -63,8 +61,23 @@ JVM_OPTIONS="$JVM_OPTIONS -Dsun.security.ssl.allowLegacyHelloMessages=false"
...
+   echo "No ALPN Agent jar found, running 'sbt stage'"
+   sbt stage;
+fi
 
+# Start `sbt` with the JVM_OPTIONS and the ALPN agent
+export AGENT=$(pwd)/$(find target -name 'jetty-alpn-agent-*.jar' | head -1)
+echo "Detected ALPN Agent: $AGENT "
 
 # Run Play (this should work for both run and start).
-SBT_OPTS=-javaagent:$AGENT $SBT $JVM_OPTIONS "$* $JVM_OPTIONS";
```
Comment (dwijnand) [inline play:70]: So does this break either run or start by not passing `"$* $JVM_OPTIONS"` anymore?

**Review by dwijnand — COMMENTED** [review]
LGTM

File: play (hunk 3 above)
Comment (octonato) [reply play:70]: that script was completely broken. 

$SBT is pointing to `sbt`

So we were setting the SBT_OPTS to:
`-javaagent:agent.jar sbt ...`

And we were not calling `sbt` at the end.

Comment (dwijnand) [reply play:70]: That's not broken that's setting an environmental variable before running a command.

SBT_OPTS was set to -javaagent:$AGENT and then sbt was being run with arguments passed.

Comment (octonato) [reply play:70]: You are right. However, the script wasn't working either. I'm don't remember the error I was getting.

Branch `2.6.x` is working, but not after the updates. So, probably related to that. 

In any case, the change proposed on this PR removes the problem with the version of the alpn and it automatically decided if it needs to run `sbt stage`.

Comment (dwijnand) [reply play:70]: > In any case, the change proposed on this PR removes the problem with the version of the alpn and it automatically decided if it needs to run `sbt stage`.

That's great.

But does it break either run or start by not passing `"$* $JVM_OPTIONS"` anymore?

File: modules/one/project/build.properties [hunk 4]
```diff
@@ -1,4 +1 @@
-#Activator-generated Properties
-#Fri Jul 08 22:56:54 PDT 2016
-template.uuid=f57599cf-a4f5-4333-9ff1-55d48fd49891
-sbt.version=1.1.0
+sbt.version=1.2.6
```
Comment (ignasi35) [inline modules/one/project/build.properties:1]: 👏🏼

File: build.sbt [hunk 5]
```diff
@@ -11,9 +11,8 @@ lazy val root = (project in file("."))
   .aggregate(one, two)
   .dependsOn(one, two)
 
-scalaVersion := "2.12.4"
-
-crossScalaVersions := Seq("2.11.12", "2.12.4")
+crossScalaVersions := Seq("2.12.7", "2.11.12")
```
Comment (ignasi35) [inline build.sbt]: Do we need to maintain the `crossScalaVersions`? Actually, why do we need it in the first place?

File: .travis.yml [hunk 6]
```diff
@@ -27,8 +27,13 @@ before_cache:
 matrix:
   exclude:
     - scala: 2.11.12
-      jdk: oraclejdk9
-
+      jdk: openjdk11
```
Comment (ignasi35) [inline .travis.yml]: The build adds and then excludes `2.11.12`. Can we get rid of it altogether?

File: README.md [hunk 7]
```diff
@@ -45,7 +45,7 @@ $ sudo vi /etc/hosts
 
 Now that you've generated the certificates and added the `example.com` host entries to `/etc/hosts`, you can start Play itself.
 
-This application is not run with `activator` -- you should run it with `./play` instead, as there are a number of system properties required to use it effectively.
+This application is not run with `sbt` -- you should run it with `./play` instead, as there are a number of system properties required to use it effectively.
```
Comment (ignasi35) [inline README.md:48]: Out of curiosity, have you tried running it with `gradle`? I have not and I wonder if it'll work or fail.

**Review by ignasi35 — COMMENTED** [review]
Left a few comments

File: README.md (hunk 7 above)
Comment (octonato) [reply README.md:48]: no, I haven't

File: .travis.yml (hunk 6 above)
Comment (octonato) [reply .travis.yml]: it excludes 2.11.12 for openkjdk11 only.

Comment (ignasi35) [reply .travis.yml]: 👍🏼

File: build.sbt (hunk 5 above)
Comment (octonato) [reply build.sbt]: that's a good point, why we are cross-building?

I looked at the history and I don't see any clear reason. I think it was just a reflex of adding it when we started to cross build to 2.11 and 2.12

Comment (ignasi35) [reply build.sbt]: Ok, I think that if it works we could _not_ remove it and add an issue to clean this up later.

Comment (ignasi35) [reply build.sbt]: Ah, just seen your new commit. LGTM too. :-)

**Comment by octonato** [comment]
I removed the cross-build. We may eventually add it back when we start to cross-build for `2.13.0-RCx`. To decide later...

File: play (hunk 3 above)
Comment (dwijnand) [reply play:70]: Not a blocker.

**Pull request merged**